### PR TITLE
Add external validation to axa-file-upload

### LINKED ISSUE #2222: File Upload - Show server validation
We'd need the posibillity for E-Forms to show the user an errormessage on the File Upload component when a file throws an error on the server validation (e.g. maleware check, to big over all components)

Our wish would be to have a function, where we just give the file-id and the error message, and then the component and the mentioned file is set to invalid

### PR DESCRIPTION
**Pull request by markus-walther**
Fixes #2222.

# Merge Checklist:
- [x] Code selfreview has been performed ([Best Practices](https://github.com/axa-ch-webhub-cloud/pattern-library/blob/develop/CONTRIBUTION.md#best-practices)).
- [x] Tests are sufficient.
- [ ] Modifications still work in IE.
- [x] Documentation is up to date.
- [x] Changelog is up to date.
- [x] Typescript typings for properties are up to date.
- [x] Designers approved changes or no approval needed.
- [x] Dependencies to other components still work.

### CODE REVIEW

File: src/components/20-molecules/file-upload/index.js [hunk 1]
```diff
@@ -34,6 +34,14 @@ const IMAGE_FILE_TYPES = 'image/jpg, image/jpeg, image/png';
 
 export const getBytesFromKilobyte = kilobyte => 1024 * kilobyte;
 
+const findFileById = (files, file, indexOnly) => {
+  for (let i = 0, n = files.length; i < n; i++) {
```
Comment (raphaellueckl) [inline src/components/20-molecules/file-upload/index.js:38]: Are there any advantages over `for (let i = 0; i < files.length; i++) {`?

File: src/components/20-molecules/file-upload/index.js [hunk 2]
```diff
@@ -34,6 +34,14 @@ const IMAGE_FILE_TYPES = 'image/jpg, image/jpeg, image/png';
...
+    if (files[i].id === file.id) {
+      return indexOnly ? i : files[i];
+    }
+  }
+  return undefined;
```
Comment (raphaellueckl) [inline src/components/20-molecules/file-upload/index.js:43]: This is the default return value, so why do you make it explicit?

File: src/components/20-molecules/file-upload/index.js (hunk 1 above)
Comment (markus-walther) [reply src/components/20-molecules/file-upload/index.js:38]: Yes, you don't evaluate the length of an array in every iteration!

File: src/components/20-molecules/file-upload/index.js (hunk 2 above)
Comment (markus-walther) [reply src/components/20-molecules/file-upload/index.js:43]: For clarity (I am going to test against undefined in line 553) and symmetry wit the other _explicit_ `return`.

**Review by raphaellueckl — APPROVED** [review]

**Pull request merged**